src/pykeen/triples/statement_factory.py

```diff
@@ -0,0 +1,600 @@
+        """Create LCWA instances for this factory's statements."""
+        s_p_q_to_multi_tails = _create_multi_label_tails_instance(
+            mapped_statements=self.mapped_statements,
+            use_tqdm=use_tqdm,
+        )
+        spq, multi_o = zip(*s_p_q_to_multi_tails.items())
+        mapped_statements: torch.LongTensor = torch.tensor(spq, dtype=torch.long)
+        labels = np.array([np.array(item) for item in multi_o], dtype=object)
+
+        # create mask
+        entity_mask = torch.zeros(self.num_entities, dtype=torch.bool)
+        entity_mask[self.non_qualifier_only_entities] = True
+
+        return LCWAInstances(
+            mapped_statements=mapped_statements,
+            entity_to_id=self.entity_to_id,
+            relation_to_id=self.relation_to_id,
+            labels=labels,
+            # entity_mask=entity_mask,
+        )
+
+    def create_slcwa_instances(self) -> SLCWAInstances:
+        """Create sLCWA instances for this factory's statements."""
+        return SLCWAInstances(
+            mapped_statements=self.mapped_statements,
+            entity_to_id=self.entity_to_id,
+            relation_to_id=self.relation_to_id,
+            entities=self.non_qualifier_only_entities,
+        )
+
+    # TODO those properties conflict with KGInfo object fields of the same name, comment for now
+    # @property
+    # def num_entities(self) -> int:
+    #     """The number of unique entities."""
+    #     return len(self.entity_to_id)
+    #
+    # @property
+    # def num_relations(self) -> int:
+    #     """The number of unique relations."""
+    #     return len(self.relation_to_id)
+
+    @property
+    def statement_length(self) -> int:
+        """The number of unique relations."""
+        return self.max_num_qualifier_pairs * 2 + 3
+
+    @property
+    def num_statements(self) -> int:
+        """The number of statements."""
+        return self.mapped_statements.shape[0]
+
+    @property
+    def qualifier_ratio(self) -> float:
+        """Return the percentage of statements with qualifiers."""
+        return (~(self.mapped_statements[:, 3::2] == self.padding_idx).all(dim=1)).sum().item() / self.num_statements
+
+    def extra_repr(self) -> str:
+        """Extra representation."""
+        return (
+            f"num_entities={self.num_entities:,}, "
+            f"num_relations={self.num_relations:,}, "
+            f"num_statements={self.num_statements:,}, "
+            f"max_num_qualifier_pairs={self.max_num_qualifier_pairs}, "
+            f"qualifier_ratio={self.qualifier_ratio:.2%}"
+        )
+
+    def __repr__(self) -> str:
+        return f"{self.__class__.__name__}({self.extra_repr()})"
+
+    @property
+    def num_triples(self) -> int:
+        return self.num_statements
+
+
+def create_statement_entity_mapping(statements: Iterable[str]):
+    entities = list(sorted(set([e for s in statements for e in s[::2]])))  # padding is already in the ndarray
```

Comment:
the comment seems to assume numpy array, but the parameter type annotation says `str`